RA_Integration/RA_Dlg_AchEditor.cpp

```diff
@@ -1681,8 +1672,34 @@ void Dlg_AchievementEditor::LoadAchievement( Achievement* pCheevo, BOOL bAttempt
 			SetDlgItemText( m_hAchievementEditorDlg, IDC_RA_BADGENAME, Widen( m_pSelectedAchievement->BadgeImageURI() ).c_str() );
 			UpdateBadge( m_pSelectedAchievement->BadgeImageURI() );
 		}
+
 		if( pCheevo->GetDirtyFlags() & Dirty_Conditions )
-			PopulateConditions( m_pSelectedAchievement );
+		{
+			HWND hCondList = GetDlgItem( m_hAchievementEditorDlg, IDC_RA_LBX_CONDITIONS );
+			if( hCondList != NULL )
+			{
+				unsigned int nGrp = GetSelectedConditionGroup();
+
+				if( ListView_GetItemCount( hCondList ) != m_pSelectedAchievement->NumConditions( nGrp ) )
+				{
+					PopulateConditions( pCheevo );
+				}
```

Comment:
update logic doesn't add or remove rows, if row count differs, fall back to recreating list.